src/memory-analyzer/gdb_api.h

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018 Author: Malte Mues
+#ifdef __linux__
+#ifndef CPROVER_MEMORY_ANALYZER_GDB_API_H
+#define CPROVER_MEMORY_ANALYZER_GDB_API_H
+#include <unistd.h>
+
+#include <exception>
+
+class namespacet;
+class symbolt;
+class irept;
+
+class gdb_apit
+{
+public:
+  explicit gdb_apit(const char *binary);
+
+  int create_gdb_process();
+  void terminate_gdb_process();
+
+  void run_gdb_to_breakpoint(const std::string &breakpoint);
+  void run_gdb_from_core(const std::string &corefile);
+
+  std::string get_value(const std::string &variable);
+  std::string get_memory(const std::string &variable);
+
+private:
+  static const int MAX_READ_SIZE_GDB_BUFFER = 600;
```

Comment:
This does worry me a little...

Reply:
Why? Or what are your concerns?

It is just the chunk size, I read from a pipe. The pipe does the real buffering in the background regarding the interprocess communication.
It is certainly interlinked with your comment regarding the stream processing.

Reply:
Fixed size buffers make me nervous in general but maybe I'm doing too much security analyses.